src/simulations/collect.jl

```diff
@@ -145,59 +178,158 @@ is not known.
 """
 collect_agent_data!(df, model, properties::Nothing, step::Int=0; kwargs...) = df
 
-function init_agent_dataframe(model::ABM, properties::AbstractArray)
-    nagents(model) < 1 && throw(ArgumentError("Model must have at least one agent to "*
-    "initialize data collection"))
+function init_agent_dataframe(model::ABM{A}, properties::AbstractArray) where {A<:AbstractAgent}
+    nagents(model) < 1 && throw(ArgumentError(
+        "Model must have at least one agent to initialize data collection",
+    ))
```

Comment:
For clarity, could you please split up this function into two? Essentially, simply split the `if std_headers` statement to just call one of two lower-level functions.

Reply:
Done for the aggregate call as well.